lib/formtastic/helpers/inputs_helper.rb

```diff
@@ -350,12 +350,21 @@ def association_columns(*by_associations) # @private
         end
       end
 
+      # Collects non-standard foreign key columns
+      def foreign_key_columns
+        if @object.present? && @object.class.respond_to?(:reflections)
+          @object.class.reflections.map {|_,v| v.options[:foreign_key] }.compact
```

Comment:
Yes formtastic's test suite is totally over-mocked and over-stubbed. I'll work over time to clean it up. 

https://github.com/justinfrench/formtastic/pull/1242 is first spark of integrating ActiveRecord.

I think you could just create some new models and add them to the schema to use real ActiveRecord objects.